server/src/main/java/com/thoughtworks/go/server/service/BuildAssignmentService.java

```diff
@@ -291,6 +292,10 @@ private Work createWork(final AgentInstance agent, final JobPlan job) {
 
                     final EnvironmentVariableContext environmentVariableContext = buildEnvVarContext(job.getIdentifier().getPipelineName());
 
+                    // Agent may have a NULL "resources" so we check and set an empty default here so that tests pass
+                    String agentResourcesAsCsv = agent.getResourceConfigs() == null ? "" : agent.getResourceConfigs().getCommaSeparatedResourceNames();
+                    environmentVariableContext.setProperty(GO_AGENT_RESOURCES, agentResourcesAsCsv, false);
+
```

Comment:
How about we don't set the property if the agent doesn't have any resources set on it.

Reply:
It actually makes more sense to do that, true. I've guarded it with an if. Downstream to our Ansible use-case we can for check for its existence and default to "None". It leads towards much stronger code. Good call. Thanks!